homeassistant/components/nest.py

```diff
@@ -51,23 +55,44 @@
 }, extra=vol.ALLOW_EXTRA)
 
 
-def request_configuration(nest, hass, config):
+async def async_nest_update_event_broker(hass, nest):
+    """
+    Dispatch SIGNAL_NEST_UPDATE to devices when nest stream API received data.
+
+    nest.update_event.wait will block the thread in most of time,
+    so specific an executor to save default thread pool.
+    """
+    _LOGGER.debug("listening nest.update_event")
+    with ThreadPoolExecutor(max_workers=1) as executor:
+        while True:
+            await hass.loop.run_in_executor(executor, nest.update_event.wait)
```

Comment:
The problem with this is that now Home Assistant is unable to shut down until an event comes in from Nest.

Reply:
Already think about it, next line will check `if hass.is_running`, and at line 172 `shut_down` method, I set the event to allow this statement continue